### PR TITLE
Refactor HueSound into a functional component

### PR DESCRIPTION
**Pull request by MonkeyDo**
Based on the userfeed-refactor branch, wait for #2844 to be merged please

### CODE REVIEW

File: frontend/js/src/explore/huesound/HueSound.tsx [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+      .catch((err) => {
+        toast.error(
+          <ToastMsg
+            title="Error"
+            message={err.message ? err.message.toString() : err.toString()}
+          />,
+          { toastId: "error" }
+        );
+      });
+  }, [selectedColorString, APIService]);
```
Comment (anshg1214) [inline frontend/js/src/explore/huesound/HueSound.tsx:55]: It is not a good idea to call the API when the component mounts. Whenever the page mounts, the releases corresponding to hex `000000` are loaded on the screen. Instead the behaviour should be similar to how it is right now on prod.

image

Comment (anshg1214) [inline frontend/js/src/explore/huesound/HueSound.tsx:55]: Set the loading state to false once the API call have been made
```suggestion
    setLoading(false);
```

File: frontend/js/tests/explore/huesound/HueSound.test.tsx [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+describe("HueSound", () => {
+  let server: SetupServerApi;
+  beforeAll(async () => {
+    // Mock the server responses
+    const handlers = [
+      http.get("/1/explore/color/000000", () =>
+        HttpResponse.json({ payload: { releases: [release] } })
+      ),
+    ];
```
Comment (anshg1214) [inline frontend/js/tests/explore/huesound/HueSound.test.tsx:51]: Again, I don't believe that it should call hex `000000` on mount, so the test in this case needs to be refactored accordingly

File: frontend/js/tests/explore/huesound/HueSound.test.tsx [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+    renderWithProviders(<HueSound />);
+    const canvasElement = screen.getByTestId("colour-picker");
+    await user.click(canvasElement);
+    const releaseButton = screen.getByRole("button");
+    await user.click(releaseButton);
+    const links = screen.getAllByRole("link");
+    expect(links.at(0)).toHaveTextContent("Prelude to Heliogabalus");
+    expect(links.at(1)).toHaveTextContent("Rorcal & Music For The Space");
+    
+    const brainzPlayerMessages = messages.filter(
```
Comment (anshg1214) [inline frontend/js/tests/explore/huesound/HueSound.test.tsx:97]: Some whitespace here
```suggestion

```

**Review by anshg1214 — CHANGES_REQUESTED** [review]
Thanks! The PR looks really good, just some small changes left.

File: frontend/js/src/explore/huesound/HueSound.tsx (hunk 1 above)
Comment (MonkeyDo) [reply frontend/js/src/explore/huesound/HueSound.tsx:55]: That was...not on purpose. Didn't realize.

Now that I look at it, I wonder if this should be a random (or pseudo-random) color on every start?
Anyway, will fix it as on prod

**Comment by MonkeyDo** [comment]
Note that this component is refactored to play nice with react-router, but has not been refactored to be used with the new BrainzPlayer with queues

**Review by anshg1214 — APPROVED** [review]
LGTM 🚀

**Pull request merged**